Create ui-c8y-1023-94-0-new-scada-widget-10388-graft-release-cd-12323.md (#4840)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-94-0-new-scada-widget-10388-graft-release-cd-12323.md

---
date: ""
title: SCADA widget enhanced
product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-61471
version: 1023.94.0
---
The "SCADA" widget has been significantly enhanced. The new version is built on modern web component technology with Lit syntax,
replacing the previous AngularJS-based approach. Key improvements include:

> - **AI-assisted SVG generation** — analyzes the selected asset's properties, measurements, alarms, and events to propose visualizations.

Reviewer: ```suggestion
- **AI-assisted SVG generation** — analyzes the selected asset's properties, measurements, alarms, and events to propose visualizations and map necessary data to placeholders.
```

> - **AI-assisted SVG generation** — analyzes the selected asset's properties, measurements, alarms, and events to propose visualizations and map necessary data to placeholders.

Reviewer: ```suggestion
- **AI-assisted SVG generation** — analyzes the selected asset's properties, measurements, alarms, and events to propose visualizations or generate one based on user's instructions, and maps corresponding data to placeholders.
```

- **Interactive placeholder mapping** — click directly on `text` or `tspan` elements in the widget preview to convert them into dynamic placeholders.
- **Flexible display settings** — control how the SVG fits within the widget, with alignment options.
- **Advanced editor** — for complex visualizations using Lit syntax and direct web component editing.
- **Legacy compatibility** — existing "SCADA" widgets using AngularJS syntax continue to work without any changes.
- **SVG migration guide** — the documentation includes a migration guide for adapting existing SVGs to Lit syntax if you want to reuse them in the new widget.

This enhancement is available as a preview feature. To enable it, open the **Preview features** page from the right drawer and turn on

> the corresponding option. For full details on configuration, SVG preparation, and migration from the legacy widget, see [SCADA 

Reviewer: ```suggestion
the corresponding option. For details on new configuration options and SVG with Lit syntax, see [SCADA 
```

> widget](https://cumulocity.com/docs/cockpit/widgets-collection/#scada) in the user documentation.

Reviewer: ```suggestion

This enhancement is available as a preview feature. To enable it, open the **Preview features** page from the right drawer and turn on
the corresponding option. For details on new configuration options and SVG with Lit syntax, see [SCADA 
widget](https://cumulocity.com/docs/cockpit/widgets-collection/#scada) in the user documentation.

Existing "SCADA" widgets using AngularJS syntax continue to work without any changes in the legacy mode. If you want to reuse existing SVGs in the new
  widget, the documentation includes a migration guide for adapting them to Lit syntax.
```

---
date: ""

> title: SCADA widget enhanced

Reviewer 2: ```suggestion
title: Enhanced SCADA widget
```

> title: New SCADA widget (#10388) [GRAFT][release/cd] (#12323)

Reviewer: ```suggestion
title: SCADA widget enhanced
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-61471
version: 1023.94.0
---
# Backport

This will backport the following commits from `develop` to `release/cd`:
- [feat(Cockpit): [MTM-61471] New SCADA widget
(#10388)](https://github.com/Cumulocity-IoT/cumulocity-ui/pull/10388)

<!--- Backport version: 12.0.4 -->

### Questions ?
Please refer to the [Backport tool
documentation](https://github.com/sorenlouv/backport)

[MTM-61471]:

> https://cumulocity.atlassian.net/browse/MTM-61471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Reviewer: ```suggestion
The "SCADA" widget has been significantly enhanced. The new version is built on modern web component technology with Lit syntax,
replacing the previous AngularJS-based approach. Key improvements include:

- **AI-assisted SVG generation** — analyzes the selected asset's properties, measurements, alarms, and events to propose visualizations.
- **Interactive placeholder mapping** — click directly on `text` or `tspan` elements in the widget preview to convert them into dynamic placeholders.
- **Flexible display settings** — control how the SVG fits within the widget, with alignment options.
- **Advanced editor** — for complex visualizations using Lit syntax and direct web component editing.
- **Legacy compatibility** — existing "SCADA" widgets using AngularJS syntax continue to work without any changes.
- **SVG migration guide** — the documentation includes a migration guide for adapting existing SVGs to Lit syntax if you want to reuse them in the new widget.

This enhancement is available as a preview feature. To enable it, open the **Preview features** page from the right drawer and turn on
the corresponding option. For full details on configuration, SVG preparation, and migration from the legacy widget, see [SCADA 
widget](https://cumulocity.com/docs/cockpit/widgets-collection/#scada) in the user documentation.
```

> - **Advanced editor** — for complex visualizations using Lit syntax and direct web component editing.

Reviewer: ```suggestion
- **Advanced editor** — for complex visualizations using Lit syntax and direct web component editing without a need to switch to an external editor.
```

> widget](https://cumulocity.com/docs/cockpit/widgets-collection/#scada) in the user documentation.

Reviewer 2: ```suggestion
widget](/cockpit/widgets-collection/#scada) in the user documentation.
```

>   widget, the documentation includes a migration guide for adapting them to Lit syntax.

Reviewer: ```suggestion
Existing "SCADA" widgets using AngularJS syntax continue to work without any changes in the legacy mode. If you want to reuse existing SVGs in the new
  widgets, the documentation includes a migration guide for adapting them to Lit syntax.
```

>   widgets, the documentation includes a migration guide for adapting them to Lit syntax.

Reviewer 2: ```suggestion
  widgets, the documentation includes migration instructions for adapting them to Lit syntax.
```
